Microsoft.Identity.Web/TokenCacheProviders/MsalAbstractTokenCacheProvider.cs

```diff
@@ -69,7 +70,12 @@ private string CacheKey
                 }
                 else
                 {
-                    return _httpContextAccessor.HttpContext.User.GetMsalAccountId();
+                    // In the case of Web Apps, the cache key is the user account Id, and the expectation is that AcquireTokenSilent
+                    // should return a token otherwise this might require a challenge
+                    // In the case Web APIs, the token cache key is a hash of the access token used to call the Web API
+                    JwtSecurityToken jwtSecurityToken = _httpContextAccessor.HttpContext.GetTokenUsedToCallWebAPI();
+                    return (jwtSecurityToken != null) ? jwtSecurityToken.RawSignature
```

Comment:
`jwtSecurityToken.RawSignature` is not a hash of the original access token, as the comment says.

Reply:
@bgavrilMS : what matters is that the key is different when the JWT is different. Do we agree that we can use the RawSignature as a key for the token cache? even if it's not a hash?
cc: @jennyf19 .

Reply:
Sure